core/include/cubos/core/metrics.hpp

```diff
@@ -0,0 +1,124 @@
+/// @file
+/// @brief Metrics and profiling utilities and macros.
+/// @ingroup core
+
+#pragma once
+
+#include <algorithm>
+#include <chrono>
+#include <mutex>
+#include <string>
+
+#include <cubos/core/api.hpp>
+#include <cubos/core/reflection/reflect.hpp>
+
+namespace cubos::core
+{
+    /// @brief Represents a metric.
+    struct Metric
+    {
+        CUBOS_REFLECT;
+        std::string mName; ///< The name of the metric.
```

Comment:
At this point you don't need to store this anymore, as the values are stored in a map with names as keys anyways.
Thus, you can just keep a `std::deque` of doubles, which should be faster. Which means that, maybe there's no need at all to have this type.